authors/apps/articles/views.py

```diff
@@ -119,3 +124,78 @@ class ArticleRetrieveBySlugAPIView(generics.RetrieveAPIView):
     renderer_classes = (ArticleRenderer,)
     permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
     lookup_field = 'slug'
+
+
+class LikeAPIView(generics.GenericAPIView):
+    """
+    post: Create an  like or dislike.
+    """
+    serializer_class = LikeSerializer
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    renderer_classes = (LikeStatusRenderer,)
+    lookup_field = 'pk'
+
+    def post(self, request, *args, **kwargs):
+        if not Article.objects.filter(pk=kwargs['pk']).exists():
+            raise NotFound(detail="Article Not found", code=404)
```

Comment:
No need to pass `code=404`. This is already provided for you by the exception

Reply:
Done